### PR TITLE
Support Model in registry in pipeline

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by YingChen1996** [comment]
It seems registry test with local component will raise 'Unable to find a record for the request' in playback mode, even if re-record. skip test for now

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py [hunk 1]
```diff
@@ -35,3 +37,105 @@ def sample_pipeline():
         pipeline_job = sample_pipeline()
         assert registry_client.jobs.validate(pipeline_job).passed
         # TODO: add test for pipeline job create with registered component on registry after support is ready on canary
+
+    @pytest.mark.skip(reason="request body still exits when re-record and will raise error "
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py:41]: let's sync offline on this. I don't quite get here

Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py:41]: We need to have a task to track this if skip

File: sdk/ml/azure-ai-ml/tests/recordings/dsl/e2etests/test_dsl_pipeline_on_registry.pyTestDSLPipelineOnRegistrytest_pipeline_with_local_component_and_registry_model_as_input.json [hunk 2]
```diff
@@ -0,0 +1,818 @@
+{
+  "Entries": [
+    {
+      "RequestUri": "https://eastus.api.azureml.ms/registrymanagement/v1.0/registries/testFeed/discovery",
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/recordings/dsl/e2etests/test_dsl_pipeline_on_registry.pyTestDSLPipelineOnRegistrytest_pipeline_with_local_component_and_registry_model_as_input.json:4]: If testFeed is having issue, let's use sdk-canary

Comment (YingChen1996) [reply sdk/ml/azure-ai-ml/tests/recordings/dsl/e2etests/test_dsl_pipeline_on_registry.pyTestDSLPipelineOnRegistrytest_pipeline_with_local_component_and_registry_model_as_input.json:4]: I think it's not problem of testFeed. It's a record problem of registry. It seems all registry relevant tests have been skip if in playbackmode.
image

Comment (YingChen1996) [reply sdk/ml/azure-ai-ml/tests/recordings/dsl/e2etests/test_dsl_pipeline_on_registry.pyTestDSLPipelineOnRegistrytest_pipeline_with_local_component_and_registry_model_as_input.json:4]: I think @elliotzh has mentioned this issue. but this workaround doesn't work for registry
image

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py (hunk 1 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py:41]: Let's track using the feature item.

Comment (YingChen1996) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline_on_registry.py:41]: use this item to track: https://msdata.visualstudio.com/Vienna/_workitems/edit/2045745

**Pull request merged**